src/back/hlsl/writer.rs

```diff
@@ -789,6 +790,9 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
 
                     // Write argument name. Space is important.
                     write!(self.out, " {}", argument_name)?;
+                    if let TypeInner::Array { size, .. } = *ty_inner {
```

Comment:
Shouldn't this be matching on the after-pointer resolved type because you could have a pointer to an array?

Reply:
yes, that sounds right

Reply:
Alright, I can follow up.